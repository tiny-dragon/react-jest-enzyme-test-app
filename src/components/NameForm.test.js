import React from 'react';
import renderer from 'react-test-renderer';
import NameForm from './NameForm';

describe('NameForm', () => {
  let component = null;

  it('renders correctly', () => {
    component = renderer.create(<NameForm />);
    /*
    Create a TestRenderer instance with the passed React element.
    It doesnâ€™t use the real DOM, but it still fully renders the component tree into memory so you can make assertions about it.
    The returned instance has the following methods and properties.
    */
  });

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
