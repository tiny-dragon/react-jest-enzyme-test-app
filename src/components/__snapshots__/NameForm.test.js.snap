// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameForm matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Name
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
